app/controllers/admin/barcode_items_controller.rb

```diff
@@ -0,0 +1,53 @@
+class Admin::BarcodeItemsController < AdminController
+  def edit
+    @barcode_item = BarcodeItem.find(params[:id])
+  end
+
+  def update
+    @barcode_item = BarcodeItem.find(params[:id])
+    if @barcode_item.update(barcode_item_params)
+      redirect_to admin_barcode_items_path, notice: "Updated Barcode Item!"
+    else
+      flash[:error] = "Failed to update this Barcode Item."
+      render :edit
+    end
+  end
+
+  def index
+    @barcode_items = BarcodeItem.all
+  end
+
+  def new
+    @barcode_item = BarcodeItem.new
+  end
+
+  def create
+    @barcode_item = BarcodeItem.create(barcode_item_params)
+    if @barcode_item.save
+      redirect_to admin_barcode_items_path, notice: "Barcode Item added!"
+    else
+      flash[:error] = "Failed to create Barcode Item."
+      render :new
+    end
+  end
+
+  def show
+    @barcode_item = BarcodeItem.includes(items: [:organization]).find(params[:id])
```

Comment:
I'm not sure how this `includes(items: [:organization])` -- do we have an organization in the admin controller?

Reply:
Maybe I just need to read up on arel stuff.

Reply:
Good catch -- that's probably vestigial. I did a lot of copypasta and so there are a few lingering bits. Hoping to catch the rest with tests. Will fix this one!